### PR TITLE
temporal-cli: fix missing meta.platforms preventing Hydra builds

### PR DESCRIPTION
**Pull request by amarshall**
At least, hopefully.

See https://github.com/NixOS/nixpkgs/issues/243087.

###### Things done

Note that x86_64-darwin build is failing with sandbox; see https://github.com/NixOS/nixpkgs/pull/242721

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
Does not work according to ofborg. Maybe you just need `meta.platforms` instead of these attributes?

**Comment by wegank** [comment]
It still has `10.rebuild-darwin : 0`, which means that there will be no new build on Hydra.

**Comment by wegank** [comment]
I still think you have to set `meta.platforms` in `symlinkJoin` manually, currently it's not done.

File: pkgs/applications/networking/cluster/temporal-cli/default.nix [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
+{ lib, fetchFromGitHub, buildGoModule, go, installShellFiles, symlinkJoin }:
 
 let
+  meta = with lib; {
+    description = "Temporal CLI";
+    homepage = "https://temporal.io";
+    license = licenses.mit;
+    maintainers = with maintainers; [ aaronjheng ];
+    mainProgram = "temporal";
+    platforms = go.meta.platforms;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/temporal-cli/default.nix:10]: For the buildGoModule that is already set, not for symlinkJoin

File: pkgs/applications/networking/cluster/temporal-cli/default.nix [hunk 2]
```diff
@@ -1,6 +1,15 @@
-{ lib, fetchFromGitHub, buildGoModule, installShellFiles, symlinkJoin }:
+{ lib, fetchFromGitHub, buildGoModule, go, installShellFiles, symlinkJoin }:
 
 let
+  meta = with lib; {
+    description = "Temporal CLI";
+    homepage = "https://temporal.io";
+    license = licenses.mit;
+    maintainers = with maintainers; [ aaronjheng ];
+    mainProgram = "temporal";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/temporal-cli/default.nix:9]: Isn't that inferred from pname

File: pkgs/applications/networking/cluster/temporal-cli/default.nix [hunk 3]
```diff
@@ -1,6 +1,15 @@
-{ lib, fetchFromGitHub, buildGoModule, installShellFiles, symlinkJoin }:
+{ lib, fetchFromGitHub, buildGoModule, go, installShellFiles, symlinkJoin }:
 
 let
+  meta = with lib; {
+    description = "Temporal CLI";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/temporal-cli/default.nix:5]: That is rather short

**Pull request merged**